packages/components/atoms/f-button/src/components/Button.vue

```diff
@@ -14,19 +14,36 @@
         v-bind="$attrs"
         :aria-live="getAriaLive"
         :aria-busy="isLoading"
-        v-on="!isLoading && $listeners"
-    >
+        v-on="!isLoading && $listeners">
         <span
             v-if="isLoading"
             :class="$style['c-spinner']"
             :data-test-id="`${componentType}-spinner`" />
-
         <span
-            :class="[
-                $style['o-btn-text'],
-                (isLoading ? $style['o-btn-text--hidden'] : '')
-            ]">
-            <slot />
+            :class="$style['o-button-content']">
+
+            <span
+                v-if="hasIcon && hasIcon === 'leading'"
```

Comment:
`hasIcon === 'leading'` should be split out into a computed property to keep logic out of the template and also to reduce duplication.

Reply:
Great suggesting, thanks, now everything looks much tidier!